### PR TITLE
Simplified simple model (moved it to new material model interface)

### PR DESCRIPTION
**Pull request by jperryhouts**
One down, eight to go.

### CODE REVIEW

**Comment by bangerth** [comment]
Yay! Let's see if the tester is happy. The rest is ok with me.

File: include/aspect/material_model/simple.h [hunk 1]
```diff
@@ -42,46 +43,14 @@ namespace aspect
...
-
-        virtual double thermal_conductivity (const double temperature,
-                                             const double pressure,
-                                             const std::vector<double> &compositional_fields,
-                                             const Point<dim> &position) const;
-        /**
-         * @}
-         */
+
+        typedef typename aspect::MaterialModel::Interface<dim>::MaterialModelInputs MaterialModelInputs;
```
Comment (tjhei) [inline include/aspect/material_model/simple.h]: We should use ```MaterialModel::MaterialModelInputs``` instead. This makes the name simple enough that you can skip declaring a typedef in my opinion. Can you try that? (```MaterialModel::Interface<dim>::MaterialModelInputs``` is just a typedef to ```MaterialModel::MaterialModelInputs``` for backwards compatibility).

File: source/material_model/simple.cc [hunk 2]
```diff
@@ -30,32 +29,55 @@ namespace aspect
...
+                                                                   1e2),
+                                                          1e-2)
+                                                 :
+                                                 1.0);
+
+          double composition_dependence = 1.0;
+          if ((composition_viscosity_prefactor != 1.0) && (in.composition[i].size() > 0))
+            {
+              //geometric interpolation
+              out.viscosities[i] = (pow(10, ((1-in.composition[i][0]) * log10(eta*temperature_dependence)
```
Comment (tjhei) [inline source/material_model/simple.cc]: can you change this from 10 to 10.0 please?

**Comment by tjhei** [comment]
good otherwise

**Comment by jperryhouts** [comment]
Changed the above, and some minor stylistic things that I found while doing that. Commits squashed. I think this is good to go, so long as the tester is okay with it.

**Comment by tjhei** [comment]
Oops, I actually have to force the tester to run it: /run-tests

**Comment by jperryhouts** [comment]
Turns out there were lots of tests which inherit from Simple. Those are updated now. Try running tests again?

File: source/material_model/simple.cc [hunk 3]
```diff
@@ -323,14 +283,8 @@ namespace aspect
...
       prm.leave_subsection();
     }
-  }
-}
 
-// explicit instantiations
-namespace aspect
-{
-  namespace MaterialModel
-  {
```
Comment (bangerth) [inline source/material_model/simple.cc]: Please leave these few lines in there. They're not strictly necessary, but they visually separate things that are logically separate.

File: source/postprocess/basic_statistics.cc [hunk 4]
```diff
@@ -44,6 +44,9 @@ namespace aspect
               const MaterialModel::Simple<dim> &material_model
                 = dynamic_cast<const MaterialModel::Simple<dim> &>(this->get_material_model());
 
+              MaterialModel::MaterialModelInputs<dim> in (1, this->n_compositional_fields());
+              MaterialModel::MaterialModelOutputs out (1, this->n_compositional_fields());
```
Comment (bangerth) [inline source/postprocess/basic_statistics.cc]: what's the purpose of these changes here? They have nothing to do with the simple material model, right?

Comment (bangerth) [inline source/postprocess/basic_statistics.cc]: Oh, I see now -- you can't call the old functions any more since they're gone? Right?

**Comment by bangerth** [comment]
Can you please rebase to current master and ensure that it compiles? There have been some changes in the meantime that will require you to adapt `MaterialModelOutputs` to `MaterialModelOutputs<dim>`.

Please leave a note when you're done.

File: source/postprocess/basic_statistics.cc (hunk 4 above)
Comment (jperryhouts) [inline source/postprocess/basic_statistics.cc]: basic_statistics.cc constructs its own material model in order to query it for values later on. For some reason it casts the actual material model to a Simple material model, and then calls functions from it. I'm removing some of those functions from the MaterialModel::Simple class, so I need to somehow still be able to query it for values. I suspect that the dynamic cast is not important anymore, but I wanted to make minimal changes.

Comment (bangerth) [inline source/postprocess/basic_statistics.cc]: Yea, I remember now. I think basic_statistics.cc eventually needs to go since it only works for some material models and not all. Anyway, thanks for clarifying.

**Comment by jperryhouts** [comment]
I think it still breaks some tests even if it does compile with the new MaterialModelOutputs template. I'll fix it after I finish the prescribed velocity cookbook and let you know when I think it's actually ready.

**Comment by jperryhouts** [comment]
Depends on  #463 but is okay otherwise.

**Comment by jperryhouts** [comment]
Finally! This turned out to be way more work than I expected.

File: source/material_model/simple.cc [hunk 5]
```diff
@@ -30,32 +29,53 @@ namespace aspect
...
+          const double delta_temp = in.temperature[i]-reference_T;
+          const double temperature_dependence = (reference_T > 0
+                                                 ?
+                                                 std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/reference_T),
+                                                                   1e2),
+                                                          1e-2)
+                                                 :
+                                                 1.0);
+
+          double composition_dependence = 1.0;
```
Comment (tjhei) [inline source/material_model/simple.cc]: what is the role of the ```composition_dependence``` variable?

File: source/material_model/simple.cc [hunk 6]
```diff
@@ -30,32 +29,53 @@ namespace aspect
...
+          out.viscosities[i] = ((composition_viscosity_prefactor != 1.0) && (in.composition[i].size() > 0))
```
Comment (tjhei) [inline source/material_model/simple.cc]: can you move ```c``` above and use it in here?

File: source/postprocess/basic_statistics.cc [hunk 7]
```diff
@@ -58,6 +61,14 @@ namespace aspect
               std::vector<double> composition_values(this->n_compositional_fields(),0.0);
 
               Point<dim> representative_point = this->get_geometry_model().representative_point(0);
+
+              in.position[0] = representative_point;
+              in.temperature[0] = dT;
+              in.pressure[0] = dT; //TODO: dT for the pressure is wrong
```
Comment (tjhei) [inline source/postprocess/basic_statistics.cc]: why use dT for pressure?

File: tests/no-viscosity-in-adiabatic-conditions.cc [hunk 8]
```diff
@@ -1,80 +0,0 @@
-#include <aspect/material_model/simple.h>
```
Comment (tjhei) [inline tests/no-viscosity-in-adiabatic-conditions.cc]: what happened to this test?

**Comment by tjhei** [comment]
looks good (but see my comments above)

File: tests/no-viscosity-in-adiabatic-conditions.cc (hunk 8 above)
Comment (jperryhouts) [inline tests/no-viscosity-in-adiabatic-conditions.cc]: The test assumed that it was possible to query the material model's viscosity independent of the other state variables. This is no longer possible with the new interface, so the test doesn't really make sense anymore. Misuse of uninitialized variables should be caught by your FPE thing now anyway.

File: source/postprocess/basic_statistics.cc (hunk 7 above)
Comment (jperryhouts) [inline source/postprocess/basic_statistics.cc]: I don't know. I just emulated the original behavior of this plugin. I have the impression this plugin isn't very good, but that's a separate issue.

Comment (bangerth) [inline source/postprocess/basic_statistics.cc]: Yes, this plugin needs to die.

File: source/material_model/simple.cc (hunk 6 above)
Comment (jperryhouts) [inline source/material_model/simple.cc]: That's probably a good idea, but it will change functionality and doesn't belong in this commit.

**Comment by bangerth** [comment]
Thanks very much! 7 more to go :-)
